### PR TITLE
Add Filter Menu in MyBooks Activity

### PR DESCRIPTION
**Pull request by huddy987**
- Add filter menu fragment to use filtered queries, this should be easily adaptable to other activities as well
- Fix a crash
- Clean up some code

### CODE REVIEW

**Comment by huddy987** [comment]
Also swapped the scan and filter buttons cuz it looks better

**Comment by huddy987** [comment]
filterMenu
filtered
newLayout

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java [hunk 1]
```diff
@@ -54,35 +50,38 @@ public BookListAdapter(Context context, FirestoreRecyclerOptions options)
...
+        holder.title.setText(book.getTitle());
+        holder.author.setText(book.getAuthor());
+        holder.isbn.setText(book.getIsbn());
 
-        book.setStatusCircleColor(book.getStatus(), bookListViewHolder.statusCircle);
+        book.setStatusCircleColor(book.getStatus(), holder.statusCircle);
 
-        holder.itemView.setOnClickListener(new View.OnClickListener()
+        /* Don't keep setting the onClickListener if it has already been set */
+        if (!holder.itemView.hasOnClickListeners())
```
Comment (KateVlaar) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/BookListAdapter.java:69]: Nice

**Review by KateVlaar — APPROVED** [review]
Can't think of any questions right now. Looks good to me

**Comment by IsaakC** [comment]
It looks like the filtering works but when clicking on the books after filtering it returns the book in that position from the unsorted list
image
image
<click on first book>
image

**Comment by huddy987** [comment]
@IsaakC Fixed

**Review by IsaakC — APPROVED** [review]

**Pull request merged**